### ISSUE TITLE
Wrong xml name in SourcedVmTemplateParams -> VmGeneralParams

### ISSUE THREAD

**Issue by coudi3bm** [vmware/go-vcloud-director#421]
Invalid xml parameter name VmGeneralParams in structure SourcedVmTemplateParams

Suggestion:
```diff
type SourcedVmTemplateParams struct {
	Source                        *Reference           `xml:"Source"`                                  // A reference to an existing VM template
- 	VmGeneralParams               *VMGeneralParams     `xml:"VMGeneralParams,omitempty"`               // Specify name, description, and other properties of a VM during instantiation.
+ 	VmGeneralParams               *VMGeneralParams     `xml:"VmGeneralParams,omitempty"`               // Specify name, description, and other properties of a VM during instantiation.
	VmTemplateInstantiationParams *InstantiationParams `xml:"VmTemplateInstantiationParams,omitempty"` // Same as InstantiationParams used for VMs within a vApp
	StorageProfile                *Reference           `xml:"StorageProfile,omitempty"`                // A reference to a storage profile to be used for the VM. The specified storage profile must exist in the organization vDC that contains the composed vApp. If not specified, the default storage profile for the vDC is used.
	LocalityParams                *LocalityParams      `xml:"LocalityParams,omitempty"`                // Locality parameters provide a hint that may help optimize placement of a VM and an independent a Disk so that the VM can make efficient use of the disk.
	VmCapabilities                *VmCapabilities      `xml:"VmCapabilities,omitempty"`                // Describes the capabilities (hot swap, etc.) the instantiated VM should have.
}
```

VMware docs reference:
https://vdc-repo.vmware.com/vmwb-repository/dcr-public/4cb94d1d-d9d0-49cd-b6e5-f4edebaf6753/b83c6907-dc7e-4f3c-9bc5-4060ed512f49/doc/doc///types/SourcedVmTemplateParamsType.html

### Reproduction steps

```bash
Send CreateStandaloneVMFromTemplateAsync with VmGeneralParams
```

**Comment by dataclouder** [comment]
Thanks for submitting this issue. I have verified it and found that the suggested fix works as expected.

**Comment by dataclouder** [comment]
Fixed in PR #420
